bigtop-packages/src/common/spark/spark-thriftserver.svc

```diff
@@ -45,7 +45,8 @@ start() {
     LOG_FILE=/var/log/spark/${DAEMON}.out
 
     runuser -s /bin/bash $SVC_USER -c "nohup nice -n 0 \
-        ${EXEC_PATH} --class org.apache.spark.sql.hive.thriftserver.HiveThriftServer2 1 $DAEMON_FLAGS \
+        ${EXEC_PATH} --class org.apache.spark.sql.hive.thriftserver.HiveThriftServer2 \
+        --hiveconf hive.server2.thrift.port=12000 1 $DAEMON_FLAGS \
```

Comment:
The port number should not be hardcoded. We usually use service default file (/etc/default/spark-thriftserver) for this. init script generated from template [loads the service default file.](https://github.com/apache/bigtop/blob/60509a922055072223639b268feb0e212666ec52/bigtop-packages/src/templates/init.d.tmpl#L190-L193)

Reply:
thanks, I fix this